protocol/contracts/dydx-messages-example/src/contract.rs

```diff
@@ -0,0 +1,54 @@
+use cosmwasm_std::to_json_binary;
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::{
+    entry_point, to_binary, Addr, BankMsg, Binary, Coin, Deps, DepsMut, Env, MessageInfo, Response,
+    StdResult
+};
+
+use crate::error::ContractError;
+use crate::msg::{ArbiterResponse, ExecuteMsg, InstantiateMsg};
+use crate::state::{Config, CONFIG};
+use cw2::set_contract_version;
+use dydx_cosmwasm::DydxMsg;
+use dydx_cosmwasm::{DydxQuerier, DydxQueryWrapper, SubaccountId};
+use dydx_cosmwasm::DydxQuery;
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:dydx-messages-example";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+    Ok(Response::default())
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response<DydxMsg>, ContractError> {
+    match msg {
+        ExecuteMsg::DydxMsg(dydx_msg) => {
+            Ok(Response::new().add_message(dydx_msg))
+        }
+    }
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn query(deps: Deps<DydxQueryWrapper>, _env: Env, msg: DydxQuery) -> StdResult<Binary> {
```

Comment:
Is this contract meant as an example for contract devs to use as a reference? I worry accepting the `DydxQuery` type from the bindings lib would be confusing to them. It's more typical for a contract to define its own `QueryMsg`.